homeassistant/components/sensor/phicomm.py

```diff
@@ -57,22 +55,29 @@
 })
 
 
-@asyncio.coroutine
-def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
     """Set up the Phicomm sensor."""
-    name = config.get(CONF_NAME)
-    username = config.get(CONF_USERNAME)
-    password = config.get(CONF_PASSWORD)
-    sensors = config.get(CONF_SENSORS)
-    scan_interval = config.get(CONF_SCAN_INTERVAL)
+    name = config[CONF_NAME]
+    username = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+    sensors = config[CONF_SENSORS]
+    scan_interval = config[CONF_SCAN_INTERVAL]
 
     phicomm = PhicommData(hass, username, password)
```

Comment:
Since this should be broken out of hass you should also have it decoupled from hass; you only need to pass the loop, session object and token path